### ISSUE TITLE
Wrong HTTP headers sent by SPARQLStore - cannot work properly with Sesame triplestore

### ISSUE THREAD

**Issue by tfrancart** [SemanticMediaWiki/SemanticMediaWiki#702]
In extensions/SemanticMediaWiki/includes/sparql/SMW_SparqlDatabase.php, method doQuery :

    curl_setopt( $this->m_curlhandle, CURLOPT_URL, $this->m_queryEndpoint );
    curl_setopt( $this->m_curlhandle, CURLOPT_HTTPHEADER, array('Accept: application/sparql-results+xml,application/xml;q=0.8' ));
    curl_setopt( $this->m_curlhandle, CURLOPT_POST, true );
    $parameterString = "query=" . urlencode( $sparql ) .
    ( ( $this->m_defaultGraph !== '' )? '&default-graph-uri=' . urlencode( $this->m_defaultGraph ) : '' );
    curl_setopt( $this->m_curlhandle, CURLOPT_POSTFIELDS, $parameterString );
    curl_setopt( $this->m_curlhandle, CURLOPT_HTTPHEADER, array('Content-Type: application/x-www-form-urlencoded;charset=UTF-8'));

    $xmlResult = curl_exec( $this->m_curlhandle );

CURLOPT_HTTPHEADER cannot be set 2 times. Thus the first header ("Accept: application/sparql-results+xml,application/xml;q=0.8'") is not send correctly. Code should be :

    curl_setopt( $this->m_curlhandle, CURLOPT_URL, $this->m_queryEndpoint );
    curl_setopt( $this->m_curlhandle, CURLOPT_HTTPHEADER, array(
         'Accept: application/sparql-results+xml,application/xml;q=0.8',
         'Content-Type: application/x-www-form-urlencoded;charset=UTF-8'
    ));

    curl_setopt( $this->m_curlhandle, CURLOPT_POST, true );
    $parameterString = "query=" . urlencode( $sparql ) .
    ( ( $this->m_defaultGraph !== '' )? '&default-graph-uri=' . urlencode( $this->m_defaultGraph ) : '' );
    curl_setopt( $this->m_curlhandle, CURLOPT_POSTFIELDS, $parameterString );

    $xmlResult = curl_exec( $this->m_curlhandle );

**Comment by mwjames** [comment]
> In extensions/SemanticMediaWiki/includes/sparql/SMW_SparqlDatabase.php, method doQuery :

`SMW_SparqlDatabase` no longer exists (in SMW 2.0/2.1), please verify against the current master [0].

[0] https://github.com/SemanticMediaWiki/SemanticMediaWiki/tree/master/includes/src/SPARQLStore

**Comment by mwjames** [comment]
As I understand, you mean? (now in SMW 2+)

```
$this->httpRequest->setOption( CURLOPT_HTTPHEADER, array(
'Accept: application/sparql-results+xml,application/xml;q=0.8',
'Content-Type: application/x-www-form-urlencoded;charset=UTF-8'
) );
```

Which Sesame triplestore you are referring to?

**Comment by tfrancart** [comment]
1. Sorry, I am using SemanticBunble which probably does not include the latest version of SMW
2. I am referring to a Sesame triplestore synchronized with SMW :

    $smwgDefaultStore = 'SMWSparqlStore';
    $smwgSparqlQueryEndpoint = 'http://127.0.0.1:8080/openrdf-sesame/repositories/smw-coopaxis';
    $smwgSparqlUpdateEndpoint = 'http://127.0.0.1:8080/openrdf-sesame/repositories/smw-coopaxis/statements';
    $smwgSparqlDataEndpoint = '';

3. Yes, I think the fix you propose in the latest codebase is correct. line 428 should be replaced by 

    $this->httpRequest->setOption( CURLOPT_HTTPHEADER, array(
      'Accept: application/sparql-results+xml,application/xml;q=0.8',
      'Content-Type: application/x-www-form-urlencoded;charset=UTF-8'
    ) );

and line 435 should be deleted.

**Comment by mwjames** [comment]
> Sorry, I am using SemanticBunble which probably does not include the latest version of SMW

This is fine but maybe in future you could tests against SMW 2.0/2.1, `SPARQLStore` comes with some #ask query features and internal performance improvements.

> I am referring to a Sesame triplestore synchronized with SMW :

We constantly test against Virtuoso and Fuseki but I'm not sure how easy it would be to add openrdf-sesame to Travis therefore having someone the confirm the Sesame compliance would be nice.

**Comment by JeroenDeDauw** [comment]
2.1 milestone??

**Comment by mwjames** [comment]
> 2.1 milestone??

Fixed in 2.1 by #703.

**Comment by mwjames** [comment]
> not sure how easy it would be to add openrdf-sesame to Travis

Fixed with #707.
